wfe2/wfe.go

```diff
@@ -530,13 +534,256 @@ func (wfe *WebFrontEndImpl) regHoldsAuthorizations(ctx context.Context, regID in
 	return !missingNames, nil
 }
 
-// RevokeCertificate is used by clients to request the revocation of a cert.
-func (wfe *WebFrontEndImpl) RevokeCertificate(ctx context.Context, logEvent *requestEvent, response http.ResponseWriter, request *http.Request) {
-	// RevokeCertificate is a NOP for WFEv2 at the present time. It needs unique JWS
-	// validation compared to other ACME v2 endpoints.
-	wfe.sendError(response, logEvent,
-		probs.ServerInternal("RevokeCertificate is not presently implemented for ACME v2"), nil)
-	return
+// authenticateRevocation is a callback function that can be used to validate
+// that the outer JWS of a revocation request is properly authenticated. Since
+// there are two ways a revocation request can be authenticated it is the
+// responsibility of the implementation to extract the outer JWS payload (and if
+// applicable, an associated account ID). If the request is not properly
+// authenticated a problem should be returned. Note that the account ID returned
+// _may_ be 0 if the request is self-authenticated and not associated with an
+// account.
+type authenticateRevocation func(*jose.JSONWebSignature, *http.Request, context.Context, *requestEvent) ([]byte, int64, *probs.ProblemDetails)
```

Comment:
`context.Context` _generally_ comes first, but that's just a style thing.

Reply:
Oops. I think I've been breaking that convention.